File: Models/DTAR/DT_Base.cs
using System;
using FoundryRulesAndUnits.Models;

namespace IoBTMessage.Models
{
	public class DESIGN_Base
	{
		public string guid { get; set; }
		public string parentGuid { get; set; }
		public string name { get; set; }
		public string type { get; set; }
		public string url { get; set; }
		public string timeStamp { get; set; }
		public ControlParameters metadata { get; set; }
	}

	
	public interface ISystem
	{

	}


	public class DO_Searchable : DESIGN_Base
	{
		public string title { get; set; }
		public string description { get; set; }
	}

	[System.Serializable]
	public class DT_Searchable : DT_Base
	{
		public string title;
		public string description;


		public DT_Searchable() : base()
		{
		}

	}

	public class DO_QualityAssurance : DO_Searchable
	{
		public string action { get; set; }
		public string author { get; set; }
		public string componentGuid { get; set; }
	}

	[System.Serializable]
	public class DT_QualityAssurance : DT_Searchable
	{
		public string action;
		public string author;
		public string componentGuid;


		public DT_QualityAssurance() : base()
		{
		}

	}

	public class DO_Comment : DO_Searchable
	{
		public string severity { get; set; }
		public string author { get; set; }
	}

	[System.Serializable]
	public class DT_Comment : DT_Searchable
	{
		public string severity;
		public string author;

		public DT_Comment() : base()
		{
		}
		public DT_Comment OK()
		{
			severity = "OK";
			return this;
		}
		public DT_Comment Error()
		{
			severity = "Error";
			return this;
		}
		public DT_Comment Bug()
		{
			severity = "Bug";
			return this;
		}
		public DT_Comment Comment()
		{
			severity = "Comment";
			return this;
		}
		public DT_Comment MissingReference()
		{
			severity = "Missing";
			return this;
		}

	}
}

File: Models/DTAR/dT_Base.ts
export interface DT_Base {
	guid: string;
	parentGuid: string;
	name: string;
	type: string;
	timeStamp: string;
}

export interface DT_Error extends DT_Base {
	error: string;
	source: any;
}

export interface DT_QualityAssurance extends DT_Base {
	comment: string;
	action: string;
	author: string;
	componentGuid: string;
}

export interface DT_Comment extends DT_Base {
	comment: string;
	severity: string;
	author: string;
}

export interface DT_Title extends DT_Base {
	title: string;
	description: string;
	comments: DT_Comment[];
}

export interface DT_Hero extends DT_Title {
	heroImage: DT_Document;
	primaryAsset: DT_AssetReference;
	assetReferences: DT_AssetReference[];
}